test/unit/test_type.c

```diff
@@ -512,6 +512,185 @@ static bool test_type_as_string(void) {
+							"\tchar c;\n"
+							"\tstruct alb {\n"
+							"\t\tconst char *b;\n"
+							"\t\tint *const *a[][][][9];\n"
+							"\t\twchar_t (*funk)(int a, const char *b);\n"
+							"\t\ttime_t t;\n"
+							"\t} ania;\n"
+							"\tint j;\n"
+							"} multunfold;";
+static char *pretty_union_of_struct = "union alpha { struct joy bla; struct { int foo; char bar; } baz; };";
+static char *pretty_union_of_struct_multiline1 = "union alpha {\n"
+						 "\tstruct joy {\n"
+						 "\t\tint a;\n"
+						 "\t\tchar c;\n"
+						 "\t\tstruct alb ania;\n"
+						 "\t\tint j;\n"
+						 "\t} bla;\n"
+						 "\tstruct {\n"
+						 "\t\tint foo;\n"
+						 "\t\tchar bar;\n"
+						 "\t} baz;\n"
+						 "} mult1;";
+static char *pretty_union_of_struct_anon_multiline = "union alpha {\n"
+						     "\tstruct joy bla;\n"
+						     "\tstruct {\n"
+						     "\t\tint foo;\n"
+						     "\t\tchar bar;\n"
+						     "\t} baz;\n"
+						     "} anonmult;";
+static char *pretty_union_of_struct_max_multiline = "union alpha {\n"
+						    "\tstruct joy {\n"
+						    "\t\tint a;\n"
+						    "\t\tchar c;\n"
+						    "\t\tstruct alb {\n"
+						    "\t\t\tconst char *b;\n"
+						    "\t\t\tint *const *a[][][][9];\n"
+						    "\t\t\twchar_t (*funk)(int a, const char *b);\n"
+						    "\t\t\ttime_t t;\n"
+						    "\t\t} ania;\n"
+						    "\t\tint j;\n"
+						    "\t} bla;\n"
+						    "\tstruct {\n"
+						    "\t\tint foo;\n"
+						    "\t\tchar bar;\n"
+						    "\t} baz;\n"
+						    "} maxmult;";
+static char *pretty_enum = "enum MCU { IRON = 0x1001, HAWK = 0x337, DOCS = 0x1337, CAPM = 0x2077 };";
+static char *pretty_enum_multiline = "enum MCU {\n"
+				     "\tIRON = 0x1001,\n"
+				     "\tHAWK = 0x337,\n"
+				     "\tDOCS = 0x1337,\n"
+				     "\tCAPM = 0x2077\n"
+				     "} enumult;";
+
+static bool test_type_as_pretty_string(void) {
+	RzTypeDB *typedb = rz_type_db_new();
+	mu_assert_notnull(typedb, "Couldn't create new RzTypeDB");
+	mu_assert_notnull(typedb->types, "Couldn't create new types hashtable");
+	const char *dir_prefix = rz_sys_prefix(NULL);
+	rz_type_db_init(typedb, dir_prefix, "x86", 64, "linux");
+
+	char *error_msg = NULL;
+	RzType *ttype = rz_type_parse_string_single(typedb->parser, pretty_complex_const_pointer, &error_msg);
+	mu_assert_notnull(ttype, "complex const pointer type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+	char *pretty_str = rz_type_as_pretty_string(typedb, ttype, "c", RZ_TYPE_PRINT_NO_OPTS, 1);
+	mu_assert_streq(pretty_str, pretty_complex_const_pointer, "complex const pointer type is ugly");
+	free(pretty_str);
+	free(ttype);
+
+	error_msg = NULL;
+	ttype = rz_type_parse_string_single(typedb->parser, pretty_struct_array_ptr_func_ptr, &error_msg);
+	mu_assert_notnull(ttype, "struct array ptr func ptr type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, NULL, RZ_TYPE_PRINT_NO_OPTS, 2);
```

Comment:
Why 2?

Reply:
To make sure the pretty printing doesn't "over-unfold".

Reply:
Ok, makes sense. Maybe add a comment on top, just to make it clear. Or put it in the message string of the mu_assert after this line.